electricitymap/contrib/lib/models/events.py

```diff
@@ -260,13 +264,13 @@ def create(
         zoneKey: ZoneKey,
         datetime: datetime,
         source: str,
-        production: ProductionMix,
+        production: Optional[ProductionMix] = None,
         storage: Optional[StorageMix] = None,
         sourceType: EventSourceType = EventSourceType.measured,
     ) -> Optional["ProductionBreakdown"]:
         try:
             # Log warning if production has been corrected.
-            if production.has_corrected_negative_values:
+            if production is not None and production.has_corrected_negative_values:
```

Comment:
Will the instanciation of `ProductionBreadown` throw an error if the production mix is None?
Otherwise we're not checking for it being None as Viktor was suggesting